chainerx_cc/.clang-tidy

```diff
@@ -4,15 +4,17 @@ bugprone-*,\
 cert-*,\
 cppcoreguidelines-*,\
 -cppcoreguidelines-pro-bounds-pointer-arithmetic,\
+-cppcoreguidelines-pro-type-union-access,\
```

Comment:
It is recommended to use `variant` by clang-tidy, which is not available in C++14 and we need to consider using a 3rd party library to address this properly.